cranelift/codegen/src/legalizer/boundary.rs

```diff
@@ -990,8 +1022,11 @@ fn spill_entry_params(func: &mut Function, entry: Block) {
         .iter()
         .zip(func.dfg.block_params(entry))
     {
-        if let ArgumentLoc::Stack(offset) = abi.location {
-            let ss = func.stack_slots.make_incoming_arg(abi.value_type, offset);
+        if let ArgumentPurpose::StructArgument(_) = abi.purpose {
```

Comment:
Maybe add a comment in this if's body, to indicate that the incoming argument had been added during legalization?